Sources/Controllers/ALKConversationViewController.swift

```diff
@@ -2147,6 +2173,32 @@ extension ALKConversationViewController: ALKConversationViewModelDelegate {
     @objc open func loadingStarted() {
         activityIndicator.startAnimating()
     }
+    
+    @objc open func checkRestrictedMesssaging() {
+        isChatBarResticted = checkRestriceted()
+    }
+    
+    @objc open func checkRestriceted() -> Bool {
+        let channelKey = viewModel.channelKey
+        let channelService = ALChannelService()
+        let channel = channelService.getChannelByKey(channelKey)
+        let whatsappSource = ["WHATSAPPCLOUDAPI", "WHATSAPPTWILIO", "WHATSAPPDIALOG360"]
+        if let platformSource = channel?.platformSource,
+           let lastMessageTime = viewModel.getLastReceivedMessage()?.createdAtTime,
+           Double(truncating: lastMessageTime) <= twentyFourHoursAgoTimeStamp(),
+           whatsappSource.contains(platformSource) {
+            chatBar.textView.text = "The last message received from this contact was over 24 hours ago. To send Template message, go to the dashboard."
```

Comment:
Don't add static value to text, add a string parameter to this function so that this function can be reused in future.

Reply:
You can reuse this by adding Else if block.